File: src/learning_time_cpp/src/time_demo.cpp
#include "rclcpp/rclcpp.hpp"

class TimeDemo : public rclcpp::Node {
public:
    TimeDemo() : Node("time_demo_cpp") {
        rclcpp::Time t1(2, 500000000L);
        rclcpp::Time t2 = this->now();

        RCLCPP_INFO(this->get_logger(), "s = %.2f, ns = %ld", t1.seconds(), t1.nanoseconds());
        RCLCPP_INFO(this->get_logger(), "s = %.2f, ns = %ld", t2.seconds(), t2.nanoseconds());
    }
private:
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<TimeDemo>();
    rclcpp::shutdown();

    return 0;
}

File: src/learning_time_py/learning_time_py/time_demo.py
import rclpy
from rclpy.node import Node
from rclpy.time import Time

class TimeDemo(Node):
    def __init__(self):
        super().__init__("time_demo_py")
        t1 = Time(seconds = 2, nanoseconds = 500000000)
        t2 = self.get_clock().now()

        self.get_logger().info(
            f"s = {t1.seconds_nanoseconds()[0]}, ns = {t1.seconds_nanoseconds()[1]}"
        )
        self.get_logger().info(
            f"s = {t2.seconds_nanoseconds()[0]}, ns = {t2.seconds_nanoseconds()[1]}"
        )

def main():
    rclpy.init()
    node = TimeDemo()
    rclpy.shutdown()

if __name__ == "__main__":
    main()
